lib/puppet/provider/ini_setting/ruby.rb

```diff
@@ -35,7 +35,14 @@ def self.namevar(section_name, setting)
   end
 
   def exists?
-    !ini_file.get_value(section, setting).nil?
+    if ini_file.has_section(section)
+      !ini_file.get_value(section, setting).nil?
+    elsif !resource[:force_new_section_creation]
+        resource[:ensure] = :absent
+        resource[:force_new_section_creation]
+    else        
+      !resource[:force_new_section_creation]
```

Comment:
Would this be idempotent? It seems like it would be trying to call create every puppet run.